lib/tls.js

```diff
@@ -137,11 +141,17 @@ function unfqdn(host) {
   return host.replace(/[.]$/, '');
 }
 
+// String#toLowerCase() is locale-sensitive so we use
+// a conservative version that only lowercases A-Z.
+function toLowerCase(c) {
+  return StringFromCharCode(32 + StringPrototypeCharCodeAt(c, 0));
+}
+
 function splitHost(host) {
-  // String#toLowerCase() is locale-sensitive so we use
-  // a conservative version that only lowercases A-Z.
-  const replacer = (c) => String.fromCharCode(32 + c.charCodeAt(0));
-  return unfqdn(host).replace(/[A-Z]/g, replacer).split('.');
+  return StringPrototypeSplit(
+    StringPrototypeReplace(unfqdn(host), /[A-Z]/g, toLowerCase),
+    '.'
+  );
```

Comment:
@ljharb @Trott I made the change, but I don't really like how the code looks like (that's why I'm currently only pushing for use of static primordials)...

Reply:
instance method primordials are used often in the node codebase; imo "how it looks" is only relevant once robustness against user code is assured.